### PR TITLE
Download windows deps from host

### PR DESCRIPTION
**Pull request by jmartin-tech**
Windows only VM build improvements.

Similar to changes previously published for [metasploitable3](https://github.com/rapid7/metasploitable3), this updates the build process for Windows VMs to download pre-requisite files needed to enable TLS 1.2 support on the build host early in the packer build process.

This incorporates limited work by @sjanusz-r7 from #61 scoped to Windows needs and refactors some of the ideas there to enable all Windows builds to complete.

Many files have moved to require TLS 1.2, to address this downloading pre-reqs that enable TLS 1.2 support on the VM. Using the host system will allow the TLS requirement to be meet even when the guest OS being built did not yet support TLS 1.2 from initial ISO images.

* updates for packer 1.9.4
* download dotnet and wmf from a provisioner run by the host OS
* update dotnet to 4.5.2
* update wmf for powershell 5.1
* adjust install process for wmf to run as SYSTEM
* move boxstarter and Ruby install into the chocolatey_installs path
* use a local list in each platform for all files
* adjust net45 install for 2008r2 quirk
* fix typo in wmf 64bit check
* update expected iso file name in old vmware tools download
* update core code to be compatible with python3
* add `requests` as direct python dep
* update docker container and build instructions to use requirements.txt

This was tested against ESXi 6.7 and should also work for VMware workstation and VMware Fusion.

### CODE REVIEW

File: build_linux_baselines.py [hunk 1]
```diff
@@ -32,6 +32,14 @@ def build_base(packer_var_file, common_vars, packerfile, replace_existing, vmSer
...
         packer_vars = json.load(packer_var_source)
 
+    # Old Packer versions used a separate iso_checksum and iso_checksum_type keys.
+    # New Packer versions combine them together: "iso_checksum_type:iso_checksum"
+    # Let's implement a workaround for old Packer versions here rather than for each individual
+    # top-level Packer json template.
+    if 'iso_checksum_type' in packer_vars:
+        packer_vars['iso_checksum'] = packer_vars['iso_checksum_type'] + ':' + packer_vars['iso_checksum']
+        del packer_vars['iso_checksum_type']
+
```
Comment (jmartin-tech) [inline build_linux_baselines.py:42]: This revision could be removed, it came in as part of migrating a commit from #61.  I theory the checksum `iso_checksum_type` is not autodetected by packer and does not really even need to be specified.

File: scripts/installs/chocolatey.ps1 [hunk 2]
```diff
@@ -40,12 +40,17 @@ function Invoke-CLR4PowerShellCommand {
...
+  New-Item -ItemType File -Path $PROFILE -Force
+}
+
 $ErrorActionPreference = 'Stop'
 Set-StrictMode -Version Latest
 
 $isWin8 = wmic os get caption | find /i '" 8 "'
 $isWin2012 = wmic os get caption | find /i '" 2012 "'
 
+$env:chocolateyVersion = '0.10.13'
```
Comment (jmartin-tech) [inline scripts/installs/chocolatey.ps1:53]: Enforces an old chocolatey version for compatibility reasons.

File: scripts/installs/vm-guest-tools.bat [hunk 3]
```diff
@@ -6,25 +6,22 @@ if %TOOLS_VER%==NEW (
...
-cmd /c ""C:\Program Files\7-Zip\7z.exe" x "C:\Windows\Temp\windows.iso" -oC:\Windows\Temp\VMWare"
+cmd /c ""C:\Program Files\7-Zip\7z.exe" x "%ISO_FILE%" -oC:\Windows\Temp\VMWare"
 
 if %OS%==64BIT (
     cmd /c C:\Windows\Temp\VMWare\setup64.exe /S /v "/qn /l*v ""%TEMP%\vmmsi.log"" REBOOT=R ADDLOCAL=ALL"
     goto :done
 )
 
 cmd /c C:\Windows\Temp\VMWare\setup.exe /S /v "/qn /l*v ""%TEMP%\vmmsi.log"" REBOOT=R ADDLOCAL=ALL"
+goto :done
```
Comment (jmartin-tech) [inline scripts/installs/vm-guest-tools.bat:25]: Masks that install result code is alway `3010` to reflect restart requirement.

File: scripts/pre_downloads/windows/download-files.sh [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  # windows dotnet 4.5 and wmf 5
+  "4a1385642c1f08e3be7bc70f4a9d74954e239317f50d1a7f60aa444d759d4f49::https://download.microsoft.com/download/6/F/5/6F5FF66C-6775-42B0-86C4-47D41F2DA187/W2K12-KB3191565-x64.msu"
+  "eb7e2c4ce2c6cb24206474a6cb8610d9f4bd3a9301f1cd8963b4ff64e529f563::https://download.microsoft.com/download/6/F/5/6F5FF66C-6775-42B0-86C4-47D41F2DA187/Win7-KB3191566-x86.zip"
+  "f383c34aa65332662a17d95409a2ddedadceda74427e35d05024cd0a6a2fa647::https://download.microsoft.com/download/6/F/5/6F5FF66C-6775-42B0-86C4-47D41F2DA187/Win7AndW2K8R2-KB3191566-x64.zip"
+  "f3430a90be556a77a30bab3ac36dc9b92a43055d5fcc5869da3bfda116dbd817::https://download.microsoft.com/download/6/F/5/6F5FF66C-6775-42B0-86C4-47D41F2DA187/Win8.1-KB3191564-x86.msu"
+  "a8d788fa31b02a999cc676fb546fc782e86c2a0acd837976122a1891ceee42c0::https://download.microsoft.com/download/6/F/5/6F5FF66C-6775-42B0-86C4-47D41F2DA187/Win8.1AndW2K12R2-KB3191564-x64.msu"
+  "6c2c589132e830a185c5f40f82042bee3022e721a216680bd9b3995ba86f3781::https://download.microsoft.com/download/E/2/1/E21644B5-2DF2-47C2-91BD-63C560427900/NDP452-KB2901907-x86-x64-AllOS-ENU.exe"
+)
+
+output_dir=../../../resources/pre_downloads/windows
```
Comment (sjanusz-r7) [inline scripts/pre_downloads/windows/download-files.sh:16]: IMO, what we refer to as 'pre-downloads' here is a bit misleading. I think having the pre-downloads be pre-downloaded with Python before calling off to Packer would be a better solution. That way, the pre-downloaded files could be made available on the guest VM's floppy/disk drive without having to do any platform-specific (Powershell & Bash) wrappers for downloading. Thoughts? 👍

Comment (jmartin-tech) [reply scripts/pre_downloads/windows/download-files.sh:16]: Actually I intentionally avoided that to allows for the the idea that packer actions could be preformed independently from the python wrapping.

Comment (jmartin-tech) [reply scripts/pre_downloads/windows/download-files.sh:16]: I am not strongly apposed to the idea, this project feeds strongly off the learnings from metasploitable3 and I kept the process consistent with the original pattern I used there.

I agree, one advantage to moving this `up` to python would be to consolidation on a single set of downloaded files. The floppy/disk drive preload does not feel all that valuable as all `resources` and `scripts` have been uploaded to the VM early. I think there may be some future learnings that could lead to making that adjustment.

Either way I would prefer to iterate on that and get this landed first as the process has been broken for some time and this PR is sufficient to remove the blocker.

File: .python-version [hunk 5]
```diff
@@ -1 +1 @@
-2.7.13
+2.7.15
```
Comment (adfoster-r7) [inline .python-version:1]: Oops, looks like I missed this PR

This line caused some things to break with the sanity tests as this pyenv version wasn't available, I think I should just be able to just install that though 🤞

Comment (adfoster-r7) [reply .python-version:1]: Ah, I see - it's actually failing over here with the docker image  `rapid7/geppetto:2019_10` having the old pyenv version installed:

https://github.com/rapid7/geppetto/blob/6c29716aba52b2cff69aa0865bda1329b8ea24ec/docker/Dockerfile#L31-L32

So it looks like we'll want to bump geppetto and publish that docker image, or revert this change for now and circle back to it separately? 🤔 

```
17:33:07 2019_10: Pulling from rapid7/geppetto
17:33:07 Digest: sha256:c921bfc8d1dbbd1ced1970e530eb8775cc81ca23347102cb370101a1a5d9e085
17:33:07 Status: Image is up to date for rapid7/geppetto:2019_10
17:33:10 pyenv: version `2.7.15' is not installed (set by /r7-source/metasploit-baseline-builder/.python-version)
```

Comment (jmartin-tech) [reply .python-version:1]: Sorry, I missed that would impact geppetto runs. The version bump should not be required as these changes do not impact the code used by geppetto. I will test rolling back just the version file for now.

**Pull request merged**